### DISCUSSION TITLE
Why is Open WebUI so big? 4.2GB

### DISCUSSION THREAD

**Discussion by faelp22** [open-webui/open-webui#6455] [General]
Does the project's Docker image already have Ollama together? why is it so big? 4.2GB

Is there a lighter version? I only need a front-end, is there a way to run just the front-end to connect to the Ollama API?

thank you for your attention

**Comment by sunjam** [comment]
Share the compose file you are looking at so it can get figured out.

**Reply by mahenning** [reply]
Ollama is only in the image of `open-webui:ollama` is used instead of `open-webui:main`. But it's not a pure frontend only app. It bundles RAG (with embedding models) and speech-to-text with whisper by default. This needs, among other things, the python torch library, which I think is a few gigabyte even without CUDA support.
The decision to include these features by default bloats the image/container, but makes it more user-friendly to "just start it", because they only have to run one container for it, and not 2-3 (or pick different images for different purposes).

**Reply by Sazzels** [reply]
ok thanks for explaining, i was wondering the same thing today. i thought ollama and the then downloaded models would take the most space, after downloading the image i was really suprised that it was so big.

```yaml
// compose file
...
image: "ghcr.io/open-webui/open-webui:v0.6.5"
...
```
```bash
// docker images
ghcr.io/open-webui/open-webui v0.6.5 ba71477c0bbc 33 hours ago 4.59GB
``` 

just to be clear, that's fine right?

**Reply by mahenning** [reply]
Yes, thats the default (no ollama/no cuda) image for the specific latest version. To just always use the latest image, you could use `:main` instead of `:v0.6.5`.

**Comment by nikhilmaddirala** [comment]
I'd also love to see a lighter version of the image that's just focused on the UI without bundling any models!

**Reply by Sohervatik** [reply]
Support that

**Comment by CallMeLaNN** [comment]
I'm using this because of Workspace feature and I just surprised that the image with cuda is 4Gb+ and without cuda is 1Gb+ (one of the biggest pulling image) still big for me because I'm not going to use torch, vision, whisper and embedding in **one box** etc. It takes a while to find the large dependencies that I might not needed. It should open for customization.

It can't be 100% local with one GPU, I mean things can get messy anyway but I notice it needs deep integration for seamless UX, let see.

Anyway I prefer external API for the quality of embedding and audio. Maybe just want to use chat first with external embedding model, no voice needed, etc... 

This is like conda vs anaconda. 

Maybe the way to go is to make more optional in the Dockerfile or build the source unless the app expected everything in place 🤔

**Reply by rgaricano** [reply]
Probably is a good option,
source is 72,6MB(backend)+7,6MB(src)
build 319 MB
& the winner:
node modules 1,6GB
(I have python lib mixed, I can't calculate)

To me, this amount of nodes could clearly be optimized, after all, it's just the rendering,
1.6GB of code is a full operative system!!!, I have smaller ones running!!
